service/controller/v14/adapter/recordsets.go

```diff
@@ -1,32 +1,37 @@
 package adapter
 
+import (
+	"github.com/giantswarm/aws-operator/service/controller/v14/key"
+	"github.com/giantswarm/microerror"
+)
+
 // The template related to this adapter can be found in the following import.
 //
 //     github.com/giantswarm/aws-operator/service/controller/v14/templates/cloudformation/guest/recordsets.go
 //
 
 type recordSetsAdapter struct {
-	APIELBHostedZones         string
-	APIELBDomain              string
-	EtcdELBHostedZones        string
-	EtcdELBDomain             string
-	IngressELBDNS             string
-	IngressELBHostedZones     string
-	IngressELBAliasHostedZone string
-	IngressELBDomain          string
-	IngressWildcardELBDomain  string
-	Route53Enabled            bool
+	BaseDomain                 string
+	GuestHostedZoneNameServers string
+	Route53Enabled             bool
 }
 
 func (r *recordSetsAdapter) getRecordSets(cfg Config) error {
```

Comment:
This is out of scope but name and signature here feels wrong. This function is actually setting values and never returns error.

Reply:
I agree this is confusing. I don't want to touch it before https://github.com/giantswarm/giantswarm/issues/3783